src/Microsoft.AspNetCore.Mvc.Core/Internal/VirtualFileResultExecutor.cs

```diff
@@ -59,14 +69,31 @@ public Task ExecuteAsync(ActionContext context, VirtualFileResult result)
                         count: null,
                         cancellation: default(CancellationToken));
                 }
-                else
+                else if (range == null || !fileStream.CanSeek)
```

Comment:
How? If I set the count for a StreamCopyOperation to be the rangeLength but cannot set the beginning of the range to the range.From.Value, it wont be the correct range. @Tratcher